### DISCUSSION TITLE
Adding a "subsystem" to log messages?

### DISCUSSION THREAD

**Discussion by Lord-Kamina** [gabime/spdlog#3197] [Q&A]
Hi. I'm currently looking into replacing a very clunky in-house logger with spdlog. In our current logger, we have "subsystems" (which are really just implemented as arbitrary strings) to mark what section of the app a message is coming from.
 With our current format we might have messages reading like this:
```
locale/debug: Checking if language is set in config.
locale/notice: Setting language to: 'en_US.UTF-8'
```
I understand I could replicate such a thing by creating a different logger for each subsystem and using the logger's name, but we really have a lot of those and I think it would end up being a tremendous mess.

I was wondering if there's any way I can just pass the subsystem in as a parameter to each logger call, and have the default log format be something like: `[%Y-%m-%d @ %T] %^*SUBSYSTEM HERE*/%l%$ -- %v`

**Comment by tt4g** [comment]
Context (PR #2907) and custom formatter (https://github.com/gabime/spdlog/wiki/3.-Custom-formatting#extending-spdlog-with-your-own-flags) may be used.

**Reply by Lord-Kamina** [reply]
MDC won't work because I want to use an async logger.

@gabime Had originally suggested just the custom flags (hadn't had time to reply) but I believe that wouldn't work either because, if I understand correctly, what that essentially does is provide a shorthand for not having to repeat the logging format in more than one place, but what I need is to be able to provide an extra parameter that I can then reference in my format string.

**Reply by tt4g** [reply]
Your idea sounds like #3176.

**Reply by Lord-Kamina** [reply]
Indeed, it seems like we want to achieve much the same thing.
